Sources/TSCBasic/Process.swift

```diff
@@ -436,16 +456,27 @@ public final class Process: ObjectIdentifierProtocol {
         var outputPipe: [Int32] = [0, 0]
         var stderrPipe: [Int32] = [0, 0]
         if outputRedirection.redirectsOutput {
-            // Open the pipes.
+            // Open the pipe.
             try open(pipe: &outputPipe)
-            try open(pipe: &stderrPipe)
-            // Open the write end of the pipe as stdout and stderr, if desired.
+
+            // Open the write end of the pipe.
             posix_spawn_file_actions_adddup2(&fileActions, outputPipe[1], 1)
-            posix_spawn_file_actions_adddup2(&fileActions, stderrPipe[1], 2)
+
             // Close the other ends of the pipe.
-            for pipe in [outputPipe, stderrPipe] {
-                posix_spawn_file_actions_addclose(&fileActions, pipe[0])
-                posix_spawn_file_actions_addclose(&fileActions, pipe[1])
+            posix_spawn_file_actions_addclose(&fileActions, outputPipe[0])
+            posix_spawn_file_actions_addclose(&fileActions, outputPipe[1])
+
+            if outputRedirection.redirectStderr {
+                // If merged was requested, send stderr to stdout.
+                posix_spawn_file_actions_adddup2(&fileActions, 1, 2)
+            } else {
+                // If no redirect was requested, open the pipe for stderr.
+                try open(pipe: &stderrPipe)
```

Comment:
This would leak the descriptors in `outputPipe` if thrown.

Reply:
that's a problem with the current code as well, no? same with any of the trys in the rest of the code (including the closes)... while I agree that that should be fixed, that particular problem is not introduced by this change, and I would consider it out of scope (since it's unclear which other things might need to be changed)

Reply:
indeed